src/network/networkHandler.ts

```diff
@@ -796,24 +757,21 @@ async function handleOnDataMessage(d: OnDataArgs, overworld: Overworld): Promise
       }
       Player.syncLobby(underworld);
       underworld.tryRestartTurnPhaseLoop();
+      underworld.progressGameState();
       underworld.assertDemoExit();
       break;
-    case MESSAGE_TYPES.SET_PLAYER_POSITION:
+    }
+    case MESSAGE_TYPES.SET_PLAYER_POSITION: {
       // This message is only for the host, it ensures that the player position
       // of the host matches exactly the player position on the player's client
       if (isHost(overworld.pie)) {
         if (fromPlayer && fromPlayer.unit && payload.x !== undefined && payload.y !== undefined) {
-          // Ensure server unit stamina matches the client's unit stamina
-          const moveDist = distance(fromPlayer.unit, payload);
-          fromPlayer.unit.stamina -= moveDist;
-          // Force set the location so that the server's unit position matches the client
-          // doing the moving
```

Comment:
Why was this removed?  I believe this fixed an important bug

Reply:
There was some multiplayer bug that was happening because of this, but I don't remember exactly what it was 😦

We wouldn't want to change a player's stamina in "SET_PLAYER_POSITION" though, right? We don't know that they walked from point A to point B, it could be a teleport or utility thing in the future. It seems like another one of those things causing weird stamina edge cases.

Reply:
This is likely the cause for the one new bug I am seeing:
> Some game actions restore stamina in headless? Always seems to set it to = 90

But I don't want to reintroduce an old bug to fix it, and I think there is probably a better approach to the current stamina issues

Reply:
This is the really tricky thing about refactors.  If code is removed and it is unclear about why it was there it will likely reintroduce regressions.
With multiplayer movement, other client's players actually move more slowly than the regular player because the MOVE_PLAYER messages don't and can't arrive instantaneously.  So to prevent them from jerking all over as their current move message is completed and it's waiting for the other one, they just move more slowly to account for the lag.  It's relatively imperceptible to players but it solves that problem.  And to ensure that they always finish at the exact same position in all clients' state, the move sends a final SET_PLAYER_POSITION.  And since they got to that position somehow, it means their stamina will have decreased in the process.
Now that I'm thinking of it, we should just have the player that moved send their final stamina along with the SET_PLAYER_POSITION message.  Let's do that.

But I do really want to emphasize: only remove code that has it's function obviously replaced by the refactor or you know exactly why it's there and that it's not needed anymore.  More: https://fs.blog/chestertons-fence/  
Otherwise we'll be chasing our tails with rare or hidden regressions.

Reply:
Implemented the stamina thing you requested.

Reply:
SET_PLAYER_POSITION now uses stamina but stamina is not sent in the payload

Reply:
Oops, sorry! I thought it was sending the unit already but it was just the Vector. Fixed.